extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCache.java

```diff
@@ -56,6 +60,24 @@ public String getName() {
         return name;
     }
 
+    @Override
+    public <K, V> Uni<V> get(K key, Function<K, V> valueLoader) {
+        Objects.requireNonNull(key, NULL_KEYS_NOT_SUPPORTED_MSG);
+        // We need to defer the CompletionStage eager computation.
+        return Uni.createFrom().deferred(new Supplier<Uni<? extends V>>() {
+            @Override
+            public Uni<? extends V> get() {
+                CompletionStage<Object> caffeineValue = getFromCaffeine(key, valueLoader);
+                return Uni.createFrom().completionStage(caffeineValue).map(new Function<Object, V>() {
+                    @Override
+                    public V apply(Object cacheValue) {
+                        return cast(cacheValue);
+                    }
+                });
+            }
+        });
+    }
+
```

Comment:
@cescoffier I need your opinion on that method. As pointed out by @emmanuelbernard [here](https://github.com/quarkusio/quarkus/pull/8631#discussion_r413783721), `CompletionStage` is an eager type and I tried to delay its computation in this code to make sure the method returns a lazy non-blocking result. Did I do things right?